src/NRediSearch/Client.cs

```diff
@@ -50,19 +50,104 @@ public enum IndexOptions
             KeepTermFrequencies = 16
         }
 
+        public sealed class IndexDefinition
+        {
+            public enum IndexType
+            {
+                /// <summary>
+                /// Used to indicates that the index should follow the keys of type Hash changes
+                /// </summary>
+                HASH
+            }
+
+            internal readonly IndexType _type = IndexType.HASH;
```

Comment:
do these fields really need to be `internal`? feels very unusual; I would *expect* `private` here; it isn't a blocker, just... feels odd

Reply:
I was just following the existing standard, the reset of the fields bellow are also marked as internal.

Reply:
fair enough :)